flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/PipelinedSubpartition.java

```diff
@@ -70,6 +70,10 @@
 	@GuardedBy("buffers")
 	private int buffersInBacklog;
 
+	/** The number of non-event buffers to be announced to the downstream. */
+	@GuardedBy("buffers")
+	private int unannouncedBacklog;
```

Comment:
The previous `buffersInBacklog` variable should be replaced by this new variable, to avoid maintaining two variables.